src/ConnectedMode/QualityProfiles/QualityProfileDownloader.cs

```diff
@@ -0,0 +1,202 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.Binding;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration;
+using SonarQube.Client;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.ConnectedMode.QualityProfiles
+{
+    internal interface IQualityProfileDownloader
+    {
+        /// <summary>
+        /// Ensures that the Quality Profiles for all supported languages are to date
+        /// </summary>
+        Task<bool> UpdateAsync(BoundSonarQubeProject boundProject, IProgress<FixedStepsProgress> progress, CancellationToken cancellationToken);
+    }
+
+    internal class QualityProfileDownloader : IQualityProfileDownloader
+    {
+        private readonly IBindingConfigProvider bindingConfigProvider;
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly IConfigurationPersister configurationPersister;
+        private readonly ISolutionBindingOperation solutionBindingOperation;
+
+        private readonly ILogger logger;
+
+        private readonly IEnumerable<Language> languagesToBind;
+
+        [ImportingConstructor]
+        public QualityProfileDownloader(
+            ISonarQubeService sonarQubeService,
+            IBindingConfigProvider bindingConfigProvider,
+            IConfigurationPersister configurationPersister,
+            ISolutionBindingOperation solutionBindingOperation,
+            ILogger logger) :
+            this(
+                sonarQubeService,
+                bindingConfigProvider,
+                configurationPersister,
+                solutionBindingOperation,
+                logger,
+                languagesToBind: Language.KnownLanguages
+                )
+        { }
+
+        internal /* for testing */ QualityProfileDownloader(
+            ISonarQubeService sonarQubeService,
+            IBindingConfigProvider bindingConfigProvider,
+            IConfigurationPersister configurationPersister,
+            ISolutionBindingOperation solutionBindingOperation,
+            ILogger logger,
+            IEnumerable<Language> languagesToBind)
+        {
+            this.bindingConfigProvider = bindingConfigProvider;
+            this.sonarQubeService = sonarQubeService;
+            this.configurationPersister = configurationPersister;
+            this.solutionBindingOperation = solutionBindingOperation;
+            this.logger = logger;
+            this.languagesToBind = languagesToBind;
```

Comment:
Uses a sub-set of the imports usesd by `BindingProcessImpl`.

The `languagesToBind` parameter is there so we can inject the list of languages to test.